docker/manager.py

```diff
@@ -183,17 +185,19 @@ def list_files(self, path):
 
         path = self._get_working_directory(path)
 
-        # `grep -v /$` matches everything that doesn't end with a
-        # trailing slash, i.e. only files since `ls -p` is used:
-        result = self.run('ls -p | grep -v /$', path)
+        # Ignore dot files (hidden files) if include_hidden is enabled:
+        predicate = '' if include_hidden else '-not -path "*/\.*"'
+        # The printf part turns './file' into 'file':
+        result = self.run('find . {0} -maxdepth 1 -type f -printf "%P\n"'.format(predicate), path)
 
         if not result.succeeded:
             if errors.FILE_NOT_FOUND_PREDICATE in result.err:
                 raise errors.DockerFileNotFoundError(path)
 
             raise errors.DockerWrapperBaseError(result.err)
 
-        return result.out.strip().split('\n')
+        out = result.out.strip()
+        return sorted(out.split('\n')) if out else []
```

Comment:
`find` produces a different ordering than what we had previously, I think sorting it before return results in the behavior we've had before.